lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -0,0 +1,1577 @@
+      case PROP_TITLE:
+        ekn_media_bin_set_title (EKN_MEDIA_BIN (object),
+                                 g_value_get_string (value));
+      break;
+      case PROP_DESCRIPTION:
+        ekn_media_bin_set_description (EKN_MEDIA_BIN (object),
+                                       g_value_get_string (value));
+      break;
+      default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+    }
+}
+
+static void
+ekn_media_bin_get_property (GObject    *object,
+                            guint       prop_id,
+                            GValue     *value,
+                            GParamSpec *pspec)
+{
+  EknMediaBin *self;
+
+  g_return_if_fail (EKN_IS_MEDIA_BIN (object));
+  self = EKN_MEDIA_BIN (object);
+
+  switch (prop_id)
+    {
+      case PROP_URI:
+        g_value_set_string (value, self->uri);
+      break;
+      case PROP_VOLUME:
+        g_value_set_double (value, gtk_scale_button_get_value (self->volume_button));
+      break;
+      case PROP_AUTOHIDE:
+        g_value_set_int (value, self->autohide);
+      break;
+      case PROP_FULLSCREEN:
+        g_value_set_boolean (value, self->fullscreen);
+      break;
+      case PROP_SHOW_STREAM_INFO:
+        g_value_set_boolean (value, self->show_stream_info);
+      break;
+      case PROP_TITLE:
+        g_value_set_string (value, self->title);
+      break;
+      case PROP_DESCRIPTION:
+        g_value_set_string (value, self->description);
+      break;
+      default:
+        G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+    }
+}
+
+#define EMB_DEFINE_ACTION_SIGNAL(klass, name, handler,...) \
+  g_signal_new_class_handler (name, \
+                              G_TYPE_FROM_CLASS (klass), \
+                              G_SIGNAL_RUN_LAST | G_SIGNAL_ACTION, \
+                              G_CALLBACK (handler), \
+                              NULL, NULL, NULL, \
+                              G_TYPE_NONE, __VA_ARGS__)
+
+static void
+ekn_media_bin_class_init (EknMediaBinClass *klass)
+{
+  GObjectClass   *object_class = G_OBJECT_CLASS (klass);
+  GtkWidgetClass *widget_class = GTK_WIDGET_CLASS (klass);
+  GtkCssProvider *css_provider;
+
+  object_class->finalize = ekn_media_bin_finalize;
+  object_class->set_property = ekn_media_bin_set_property;
+  object_class->get_property = ekn_media_bin_get_property;
+
+  widget_class->realize = ekn_media_bin_realize;
+
+  /* Properties */
+  properties[PROP_URI] =
+    g_param_spec_string ("uri",
+                         "URI",
+                         "The Media URI to playbacks",
+                         NULL,
+                         G_PARAM_READWRITE);
+
+  properties[PROP_VOLUME] =
+    g_param_spec_double ("volume",
+                         "Volume",
+                         "Stream volume",
+                         0.0, 1.0, 1.0,
+                         G_PARAM_READWRITE);
+
+  properties[PROP_AUTOHIDE] =
+    g_param_spec_int ("autohide",
```

Comment:
nitpick: Suggest calling this `autohide-timeout`, just `autohide` sounds like it should be a boolean